src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -1329,15 +1329,19 @@ namespace QgsWms
       }
       else if ( !f.isEmpty() )
       {
-        // filter format: "LayerName:filterString;LayerName2:filterString2;..."
+        // filter format: "LayerName,LayerName2:filterString;LayerName3:filterString2;..."
         // several filters can be defined for one layer
         const QStringList splits = f.split( ':' );
         if ( splits.size() == 2 )
         {
-          QgsWmsParametersFilter filter;
-          filter.mFilter = splits[1];
-          filter.mType = QgsWmsParametersFilter::SQL;
-          filters.insert( splits[0], filter );
+          QStringList layers = splits[0].contains( ',' ) ? splits[0].split( ',' ) : QStringList() << splits[0];
```

Comment:
```suggestion
          const QStringList layers = splits[0].contains( ',' ) ? splits[0].split( ',' ) : QStringList() << splits[0];
```

Reply:
Is the `?` required?

I would imagine a simple `const QStringList layer = splits.at( 0 ).split( ',' );` would produce the same, no?

Reply:
you right, I made the modifications